### DISCUSSION TITLE
Updating to newer remix version on existing project

### DISCUSSION THREAD

**Discussion by roydigerhund** [remix-run/remix#2175] [Q&A] [answered]
Hey guys, what is your workflow on updating remix to the newest version in an existing project?

I couldn't find anything about this in the docs.

**Comment by sergiodxa** [comment, marked as answer]
You need to upgrade all the `@remix-run/*` packages and the `remix` package.

And check the release notes of the new versions for what changed, but there has never been a breaking change since v1.0.0 and it shouldn't be one until v2 except for `unstable_` features.

**Reply by roydigerhund** [reply]
What about changes in files like `package.json` or `remix.config.js`? 
My current way is to create a new clean project and compare those files.

**Reply by MichaelDeBoey** [reply]
Changes to `package.json` or `remix.config.js` aren't **really** necessary to do. They're only the new recommended values.

**Reply by kiliman** [reply]
You can also look at the latest templates at https://github.com/remix-run/remix/tree/main/packages/create-remix/templates and compare them to what you have.

**Reply by roydigerhund** [reply]
Perfect, that's what I was looking for 👍

**Comment by marr** [comment]
I upgraded the individual packages from 1.2.1 to 1.3.5, and now have several errors running my application. It seems the exports from `remix` no longer exist, so I will have to replace my imports to the `@remix-run/node` and `@remix-run/react` packages.. is there a codemod to do this?

**Reply by pcattori** [reply]
@marr : We have a codemod already in the works (see #2670) 🎉 , but we expect imports from `remix` to still work for any v1 versions.

---

To populate the (deprecated) `remix` imports, Remix uses `remix setup <runtime>` in your `postinstall` script. Some package managers don't reliably run `postinstall` nor do they tend to play nicely when `remix setup` modifies the contents of `node_modules/remix/`. Which is why we're deprecating all that, but things should still work if `remix setup` gets run somehow.

Can you confirm what the contents of `node_modules/remix/` looks like for you and if running `remix setup <runtime>` (e.g. `remix setup node`) solves the issue for you?
